### PR TITLE
Use a pre-commit workflow to check style

### PR DESCRIPTION
**Pull request by fsimonis**
This PR replaces custom linting workflows with a pre-commit workflow

### CODE REVIEW

File: .github/workflows/check-scripts.yaml [hunk 1]
```diff
@@ -1,22 +0,0 @@
...
-      - master
-      - develop
-jobs:
-  check_scripts:
-    runs-on: ubuntu-latest
-    steps:
-      - name: Check out repository
-        uses: actions/checkout@v2
-      - run:  bash tools/check.sh
-      - run:  bash tools/check-size.sh
```
[the remark this answers is not in the document]
Comment (fsimonis) [reply .github/workflows/check-scripts.yaml:18]: Added them. I think we need to refactor some checks at some point.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -8,9 +8,11 @@ repos:
   rev: v0.30.0
   hooks:
     - id: markdownlint
-      files: "^(.*/)?README.md"
+      #files: "^(.*/)?README.md"
```
[the remark this answers is not in the document]
Comment (fsimonis) [reply .pre-commit-config.yaml]: Was a test

File: tools/check.sh [hunk 3]
```diff
@@ -9,7 +9,7 @@ tutorials=$(find . -maxdepth 1 -type d -not -name ".*" | grep -vE $IGNORE | sed
 
 for tutorial in $tutorials; do
   # Check permalinks
-  docs=$(find "./$tutorial" -maxdepth 1 -type f -name "*.md" -print0 | xargs grep -l "permalink:" | sed "s/^.\///")
+  docs=$(find "./$tutorial" -maxdepth 1 -type f -name "*.md" -print0 | xargs -0 grep -l "permalink:" | sed "s/^.\///")
```
Comment (MakisH) [inline tools/check.sh:12]: Why this change?

**Review by MakisH — APPROVED** [review]

File: tools/check.sh (hunk 3 above)
Comment (fsimonis) [reply tools/check.sh:12]: The script triggers warnings using an up2date xargs. Small enough fish to fry here.
